scripts/recipe_robot_lib/inspect.py

```diff
@@ -614,6 +614,29 @@ def inspect_disk_image(input_path, args, facts):
     else:
         facts["inspections"].append("disk_image")
 
+    # See if we can determine the download URL from the file metadata.
+    if "download_url" not in facts:
+        cmd = "/usr/bin/xattr -p com.apple.metadata:kMDItemWhereFroms \"%s\"" % input_path
+        exitcode, out, err = get_exitcode_stdout_stderr(cmd)
+        if exitcode == 0:
+            # Save the hex output as a file.
+            where_froms_path = os.path.join(CACHE_DIR, "kMDItemWhereFroms.plist")
+            with open(where_froms_path, "wb") as where_froms_file:
+                where_froms_file.write(out)
+            # Convert the hex to a binary plist.
```

Comment:
Using the Python xattr module would also eliminate the need to convert to binary plist -- the value returned is already the bplist